components/tabs/src/storage.rs

```diff
@@ -509,22 +593,27 @@ impl TabsStorage {
          DELETE FROM pending_remote_tab_closures
          WHERE NOT EXISTS (
              SELECT 1 FROM new_remote_tabs
-             WHERE new_remote_tabs.client_id = pending_remote_tab_closures.client_id
+             WHERE new_remote_tabs.device_id = pending_remote_tab_closures.device_id
              AND new_remote_tabs.url = pending_remote_tab_closures.url
          )";
         conn.execute(delete_sql, [])?;
+        log::info!(
+            "deleted {} pending tab closures because they were not in the new tabs",
+            conn.changes()
+        );
 
         const TWENTY_FOUR_HRS_MS: u64 = 24 * 60 * 60 * 1000; // 24 hours in ms
 
         // Anything that couldn't be removed above and is older than 24 hours
         // is assumed not closeable and we can remove it from the list
```

Comment:
Note for myself (or if you'd like to do it—either is great!) to [make the command TTL in `fxa-client` match this](https://github.com/mozilla/application-services/blob/33ea53616a4d35989059bde593a68dc424a9b95e/components/fxa-client/src/internal/commands/close_tabs.rs#L17).

Reply:
I ended up making this match fxa - I believe 24 hours is maybe too short - I'm mildly concerned 48 is too - eg, sending a close command to my work computer on a Friday night will ttl out before I restart the PC Monday - but it's fine for now and I've left a breadcrumb pointing to the other in both places.